### PR TITLE
Improve check around accumulation mode

### PR DESCRIPTION
**Pull request by tgroh**
Be sure to do all of the following to help us incorporate your contribution
quickly and easily:

 - [x] Make sure the PR title is formatted like:
   `[BEAM-<Jira issue #>] Description of pull request`
 - [x] Make sure tests pass via `mvn clean verify`. (Even better, enable
       Travis-CI on your fork and ensure the whole test matrix passes).
 - [x] Replace `<Jira issue #>` in the title with the actual Jira issue
       number, if there is one.
 - [x] If this contribution is large, please file an Apache
       [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.txt).

---
If multiple panes can ever be produced, accumulation mode must be
explicitly set, as there is currently no default accumulation mode. A
pipeline with allowed lateness and the default trigger can produce
multiple panes; as a result, accumulation mode must be set even if there
is no custom trigger if allowed lateness is specified.

### CODE REVIEW

**Comment by tgroh** [comment]
R: @kennknowles OR @bjchambers 
CC: @francesperry

File: sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/windowing/Window.java [hunk 1]
```diff
@@ -434,22 +434,36 @@ public void validate(PCollection<T> input) {
...
+      }
+    }
+
+    private boolean canProduceMultiplePanes(WindowingStrategy<?, ?> strategy) {
+      // The default trigger is Repeatedly.forever(AfterWatermark.pastEndOfWindow()); This fires
+      // for every late-arriving element if allowed lateness is nonzero, and thus we must have
+      // an accumulating mode specified
+      boolean dataCanArriveLate =
+          !(strategy.getWindowFn() instanceof GlobalWindows)
+              && strategy.isAllowedLatenessSpecified()
```
Comment (kennknowles) [inline sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/windowing/Window.java]: This check is redundant and potentially dangerous. If it is not specified, but is > 0, data can still arrive late, even though it means something else has gone sideways.

File: sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/windowing/Window.java [hunk 2]
```diff
@@ -434,22 +434,36 @@ public void validate(PCollection<T> input) {
...
+              && strategy.getAllowedLateness().getMillis() > 0;
+      boolean hasCustomTrigger =
+          strategy.isTriggerSpecified() && !(strategy.getTrigger() instanceof DefaultTrigger);
```
Comment (kennknowles) [inline sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/windowing/Window.java]: Ditto. `hasNonDefaultTrigger`.

File: sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/windowing/Window.java (hunk 1 above)
Comment (tgroh) [inline sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/windowing/Window.java]: Updated. I think this is what you're suggesting.

File: sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/windowing/Window.java (hunk 2 above)
Comment (tgroh) [inline sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/windowing/Window.java]: Likewise.

**Comment by kennknowles** [comment]
Yup, exactly. LGTM.

**Pull request closed without merging**